frontend/src/components/Forms/parties/AddRolesForm.js

```diff
@@ -0,0 +1,126 @@
+
+const panelHeader = (removeField, roleNumber) => (
+  <div className="inline-flex between">
+    <Form.Item style={{ marginTop: "15px" }} label={`Role ${roleNumber}`} />
+    <div>
+      <Popconfirm
+        placement="topRight"
+        title={`Are you sure you want to remove Role ${roleNumber}?`}
+        onConfirm={removeField(roleNumber)}
+        okText="Yes"
+        cancelText="No"
+      >
+        <Button ghost>
+          <Icon type="minus-circle" theme="outlined" style={{ color: "#BC2929" }} />
+        </Button>
+      </Popconfirm>
+    </div>
+  </div>
+);
+
+const transformMineNames = (names) =>
+  names.map(({ mine_name, guid }) => (
+    <AutoComplete.Option key={guid} value={guid}>
+      {mine_name}
+    </AutoComplete.Option>
+  ));
+
+export const AddRolesForm = (props) => (
+  <div>
+    <Form>
+      <Collapse>
+        {props.roleNumbers.map((roleNumber) => (
+          <Collapse.Panel header={panelHeader(props.removeField, roleNumber)} key={roleNumber}>
+            <Row gutter={16}>
+              <Col span={12}>
+                <Field
+                  label="Role *"
+                  id={`mine_party_appt_type_code-${roleNumber}`}
+                  name={`mine_party_appt_type_code-${roleNumber}`}
+                  placeholder="Search for Mine"
+                  component={renderConfig.SELECT}
+                  data={simpleRelationships(props.partyRelationshipTypesList)}
+                  validate={[required]}
+                />
+              </Col>
+              <Col span={12}>
+                <Form.Item label="Mine *">
+                  <Field
+                    id={`mine_guid-${roleNumber}`}
+                    name={`mine_guid-${roleNumber}`}
+                    component={renderConfig.AUTOCOMPLETE}
+                    placeholder="Please add Mine"
+                    data={transformMineNames(props.mineNameList)}
+                    handleChange={props.handleChange(roleNumber)}
+                    handleSelect={props.handleSelect(roleNumber)}
+                    iconColor="rgba(0, 0, 0, 0.25)"
+                    validate={[required]}
+                  />
+                </Form.Item>
+              </Col>
+            </Row>
+
+            <Row gutter={16}>
+              <Col span={12}>
+                <Field
+                  label="Start Date *"
+                  id={`start_date-${roleNumber}`}
+                  name={`start_date-${roleNumber}`}
+                  placeholder="yyyy-mm-dd"
+                  component={renderConfig.DATE}
+                  validate={[required]}
+                />
+              </Col>
+              <Col span={12}>
+                <Field
+                  label="End Date *"
+                  id={`end_date-${roleNumber}`}
+                  name={`end_date-${roleNumber}`}
+                  placeholder="yyyy-mm-dd"
+                  component={renderConfig.DATE}
+                  validate={[required]}
+                />
```

Comment:
Are `start_date` and `end_date` required?

Reply:
I don't know if they're supposed to be. I had made them required on Friday for whatever reason. If that's inconsistent then I will gladly remove them from the validation

Reply:
As per response in the dev channel, Looks like it was a business decision to have neither dates be required.
Can dig into this further if you want, otherwise remove the validations and the '*' from the label.